pkg/router/controller/host_admitter.go

```diff
@@ -165,14 +152,22 @@ func (p *HostAdmitter) SetLastSyncProcessed(processed bool) error {
 	return p.plugin.SetLastSyncProcessed(processed)
 }
 
+// recordRejection records why the route was rejected.
+func (p *HostAdmitter) recordRejection(route *routeapi.Route, reason, message string) {
+	oldPolicy := route.Spec.WildcardPolicy
+	route.Spec.WildcardPolicy = routeapi.WildcardPolicyNone
```

Comment:
why mess with the wildcardPolicy here? If I request host foo.com with wildcardPolicy=Subdomain, and the router rejects it, it would make sense to me to see the rejection also have host foo.com and wildcardPolicy=Subdomain